### PR TITLE
YACHT-965: Stream results to new table

### PR DESCRIPTION
**Pull request by jarekdrabek**

### CODE REVIEW

File: src/commons/big_query/streaming/data_streamer.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        return httplib2.Http(timeout=60)
+
+    def __init__(self, project_id, dataset_id, table_id):
+        self.big_query_table = BigQueryTable(project_id, dataset_id, table_id)
+        self.http = self._create_http()
+        self.service = googleapiclient.discovery.build(
+            'bigquery',
+            'v2',
+            credentials=GoogleCredentials.get_application_default(),
+            http=self.http
```
Comment (przemyslaw-jasinski) [inline src/commons/big_query/streaming/data_streamer.py]: You may inline this

File: src/commons/big_query/streaming/data_streamer.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        )
+
+    def stream_stats(self, rows, insert_id=None):
+        if insert_id is None: insert_id = uuid.uuid4()
```
Comment (przemyslaw-jasinski) [inline src/commons/big_query/streaming/data_streamer.py]: IntelliJ yells to make new line after colon. Anyway, I found that it's equivalent to this:
`insert_id = insert_id or uuid.uuid4()`

File: src/commons/big_query/streaming/data_streamer.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                          self.big_query_table)
+
+    @retry(Error, tries=2, delay=2, backoff=2)
+    def _stream_metadata(self, insert_all_data):
+        partition = datetime.datetime.now().strftime("%Y%m%d")
+        return self.service.tabledata().insertAll(
+            projectId=self.big_query_table.get_project_id(),
+            datasetId=self.big_query_table.get_dataset_id(),
+            tableId='{}${}'.format(self.big_query_table.get_table_id(), partition),
+            body=insert_all_data).execute(num_retries=3)
```
Comment (przemyslaw-jasinski) [inline src/commons/big_query/streaming/data_streamer.py]: Missing blank line in the end of the file

File: src/commons/big_query/streaming/data_streamer.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    def stream_stats(self, rows, insert_id=None):
+        if insert_id is None: insert_id = uuid.uuid4()
+        insert_all_data = {
+            'rows': [{
+                'json': data,
+                'insertId': str(insert_id)
+            } for data in rows]
+        }
+        logging.info("Streaming data to table %s (insertId:%s)", self.big_query_table, insert_id)
```
Comment (przemyslaw-jasinski) [inline src/commons/big_query/streaming/data_streamer.py]: Do we keep going with our '80 max length line' convention?

File: src/slo/x_days_sli/sli_results_streamer.py [hunk 5]
```diff
@@ -1,12 +1,22 @@
...
+from src.commons.config.configuration import configuration
 
-class SLIResultsStreamer(object):
 
-    def __init__(self, big_query):
-        self.big_query = big_query
+class SLIResultsStreamer(object):
+    def __init__(self,
+                 project_id=configuration.backup_project_id,
+                 dataset_id="SLO_views_legacy",
```
Comment (przemyslaw-jasinski) [inline src/slo/x_days_sli/sli_results_streamer.py]: It streams result to partitioned table, so it shouldn't go to dataset with 'views' in name.

File: src/slo/x_days_sli/sli_results_streamer.py [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
+                 table_id="SLI_filtered"
+                 ):
+        self.data_streamer = DataStreamer(project_id=project_id, dataset_id=dataset_id, table_id=table_id)
 
     def stream(self, sli_results):
-        logging.info("TODO here results should be streamed into BQ")
-        logging.info("SLI results: %s", sli_results)
-        pass
+        if len(sli_results)==0:
```
Comment (przemyslaw-jasinski) [inline src/slo/x_days_sli/sli_results_streamer.py]: Missing whitespace around operator

File: src/commons/big_query/streaming/data_streamer.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        insert_all_response = self._stream_metadata(insert_all_data)
+        if 'insertErrors' in insert_all_response:
+            logging.debug("Sent json: \n%s", json.dumps(insert_all_data))
+            logging.error("Error during streaming metadata to BigQuery: \n%s",
```
Comment (marcin-kolda) [inline src/commons/big_query/streaming/data_streamer.py]: Please assure that we will see this error in Error Reporting

File: src/commons/big_query/streaming/data_streamer.py (hunk 1 above)
Comment (jarekdrabek) [reply src/commons/big_query/streaming/data_streamer.py]: what do you mean?

File: src/commons/big_query/streaming/data_streamer.py (hunk 2 above)
Comment (jarekdrabek) [reply src/commons/big_query/streaming/data_streamer.py]: Done

File: src/commons/big_query/streaming/data_streamer.py (hunk 3 above)
Comment (jarekdrabek) [reply src/commons/big_query/streaming/data_streamer.py]: Done

File: src/commons/big_query/streaming/data_streamer.py (hunk 4 above)
Comment (jarekdrabek) [reply src/commons/big_query/streaming/data_streamer.py]: Done

File: src/slo/x_days_sli/sli_results_streamer.py (hunk 5 above)
Comment (jarekdrabek) [reply src/slo/x_days_sli/sli_results_streamer.py]: Let's ignore it now :)

File: src/slo/x_days_sli/sli_results_streamer.py (hunk 6 above)
Comment (jarekdrabek) [reply src/slo/x_days_sli/sli_results_streamer.py]: Done

File: src/commons/big_query/streaming/data_streamer.py (hunk 7 above)
Comment (jarekdrabek) [reply src/commons/big_query/streaming/data_streamer.py]: Done

File: src/slo/x_days_sli/sli_results_streamer.py (hunk 5 above)
Comment (marcin-kolda) [reply src/slo/x_days_sli/sli_results_streamer.py]: I would move the table to the other dataset as well, then we could have different ACLs for them, i.e. read-only for table and read/write for views

**Pull request merged**